docs/UPLOAD-UTILITY.md

```diff
@@ -0,0 +1,93 @@
+
+#### On-Device Data
+
+To upload test data that is stored on a Murakami measurement device, first login to the device itself. If this is a Balena Cloud managed device, you can use the terminal through your Balena app or the balena-cli tool to login to the main container. If this is a standalone device, SSH to the device.
+
+In either case, when you’re on the device’s terminal, first confirm that murakami-upload is a usable program in the Murakami release running on the device.
+
+```
+root@d9fa4d8:/murakami# ls scripts/
+convert.py  upload.py
+root@d9fa4d8:/murakami# which murakami-upload
+/usr/local/bin/murakami-upload
+```
+
+To learn more about how to use murakami-upload, type: `murakami-upload -h`
+
+```
+murakami-upload -h
+usage: murakami-upload [-h] [-p PATH] [-u URL]
+Uploads JSON results via HTTPExporter
+optional arguments:
+  -h, --help           show this help message and exit
+  -p PATH, -path PATH  input path
+  -u URL, -url URL     URL to send JSON data to
+```
+
+On a Balena managed device with the local exporter enabled, test result data will be in `/data/`
+
+To upload results from Nov. 9-17, 2020, we might upload all test results from all test runners, one day at a time using a command such as:
+
+`murakami-upload -p '/data/*2020-11-09*' -u '<ip address or hostname>/api/v1/runs'`
+
+When you run the murakami-upload script, the environment variables available on the device are printed in the terminal output, followed by successful status messages such as those listed below:
+
+```
+Reading path /data/*2020-11-09*
+Files: ['/data/speedtest-cli-multi-stream-mlab-testbed-home-wired-2020-11-09T17:49:02.633159.jsonl', '/data/ndt5-mlab-testbed-home-wired-2020-11-09T17:48:09.359179.jsonl', '/data/ndt7-mlab-testbed-home-wired-2020-11-09T17:48:36.805318.jsonl', '/data/speedtest-cli-single-stream-mlab-testbed-home-wired-2020-11-09T17:49:45.946055.jsonl', '/data/ndt7-mlab-testbed-home-wired-2020-11-09T17:46:34.017050.jsonl']
+```
+
+Within a few minutes of completion, the uploaded data should be available in your instance of Murakami-Viz.
+
+### Data from GCS or SCP Server Archive
+
+If you’ve exported your data to GCS or an SCP server, you can use murakami-upload from a Docker container on any computer to send it to a running instance of Murakami-Viz.
+
+First, obtain a local copy of the test data you wish to upload to your instance of Murakami-Viz. For demonstration purposes, we have created a folder called `murakami-upload` and a subfolder, `data`, on our local computer. We then copied some test result files from GCS into the `data` folder.
+
+Next, we will run a Murakami container locally (not on a test device, but on your computer), and mount the `data` folder as a volume within it. The command below is an example which you can customize for the paths to your data folder:
+
+`docker run -it --entrypoint /bin/bash --volume /path/to/test-results:/data/ measurementlab/murakami:latest`
```

Comment:
What do you think of recommending a tagged release here? Ideally, `latest` should be the latest version coming from one of the sandbox branches or master and could contain WIP code -- so, it could be temporarily broken.

Reply:
`latest` is currently updated when `master` is updated.

Reply:
That's right, but `master` is the current development branch, not a release, and I don't think people should ever use `latest` unless they want to live on the edge. 

I think tagging a release whenever we add features and/or fix bugs and we know the codebase is in a good state + recommending using release tags on the documentation would be a safer option for the general public. I know we don't have any released version of Murakami, maybe it's a good time to tag one? :slightly_smiling_face: